Tidy up warning errors

mdio/dataset.h:780:36: error: logical not is only applied to the left hand side of comparison [-Werror=logical-not-parentheses]
  780 |       if (!var.dimensions().rank() == 1) {

--- a/mdio/dataset.h
+++ b/mdio/dataset.h
@@ -777,7 +777,7 @@ class Dataset {
 
       MDIO_ASSIGN_OR_RETURN(
           auto var, variables.at(std::string(descriptor.label.label())));
-      if (!var.dimensions().rank() == 1) {
+      if (var.dimensions().rank() != 1) {
         return absl::InvalidArgumentError("Label must be 1D.");
       }
 

